velox/vector/VectorTypeUtils.h

```diff
@@ -84,5 +85,9 @@ struct TypeToFlatVector {
   using type = typename KindToFlatVector<SimpleTypeTrait<T>::typeKind>::type;
 };
 
+template <typename T, bool comparable, bool orderable>
+struct TypeToFlatVector<Generic<T, comparable, orderable>> {
+  using type = BaseVector;
```

Comment:
this is a bit confusing as the expectation here is that it should return a flat vector, however, like you pointed out earlier, there is no better alternative. Since Generic is a special case that warrants these kind of mapping which look like hacks, can you add comments here to explain these decisions? They can get verbose, but at the same time would be great to have them here as the complexity of this choice is not obvious to a new reader.

What do you think?